src/app-exception-filters/main-exception.filter.ts

```diff
@@ -0,0 +1,35 @@
+import {
+  ExceptionFilter,
+  Catch,
+  ArgumentsHost,
+  HttpException,
+} from '@nestjs/common';
+import { Request, Response } from 'express';
+
+@Catch()
+export class MainExceptionFilter implements ExceptionFilter {
+  catch(exception: Error, host: ArgumentsHost) {
+    const ctx = host.switchToHttp();
+    const response = ctx.getResponse<Response>();
+    const request = ctx.getRequest<Request>();
+    const status =
+      exception instanceof HttpException ? exception.getStatus() : 500;
+
+    this.printErrorToConsole(exception, status);
+
+    response.status(status).json({
+      statusCode: status,
+      timestamp: new Date().toISOString(),
+      path: request.url,
+    });
+  }
+
+  private printErrorToConsole(exception: Error, status: number): void {
+    console.log('\n', '>>>>> MAIN APP EXCEPTIONS HANDLER: <<<<<<');
```

Comment:
jak juz chcesz tak robic to dodaj loggera bo on ci to wszystko pokrywa https://docs.nestjs.com/techniques/logger

Reply:
moze nie wszystko ale wiekszosc ;)